btsieve/src/ethereum/web3_http_blocksource.rs

```diff
@@ -0,0 +1,60 @@
+use crate::{
+    blocksource::{BlockSource, Error},
+    web3::{
+        self,
+        futures::{Future, Stream},
+        transports::Http,
+        types::{Block, BlockId, Transaction},
+        Web3,
+    },
+};
+use ethereum_support::Network;
+use std::{sync::Arc, time::Duration};
+use tokio::timer::Interval;
+use web3::types::BlockNumber;
+
+pub struct Web3HttpBlockSource {
+    web3: Arc<Web3<Http>>,
+    network: Network,
+}
+
+impl Web3HttpBlockSource {
+    pub fn new(web3: Arc<Web3<Http>>) -> impl Future<Item = Self, Error = web3::Error> {
+        web3.clone().net().version().map(move |version| Self {
+            web3,
+            network: Network::from_network_id(version),
+        })
+    }
+}
+
+impl BlockSource for Web3HttpBlockSource {
+    type Block = Block<Transaction>;
+    type Error = web3::Error;
+
+    fn blocks(&self) -> Box<dyn Stream<Item = Self::Block, Error = Error<Self::Error>> + Send> {
+        let web = self.web3.clone();
+
+        let poll_interval = match self.network {
+            Network::Mainnet => 5,
+            Network::Ropsten => 5,
+            Network::Regtest => 1,
+            Network::Unknown => 1,
+        };
+
+        log::info!(target: "ethereum::blocksource", "polling for new blocks on {} every {} seconds", self.network, poll_interval);
+
+        let stream = Interval::new_interval(Duration::from_secs(poll_interval))
+            .map_err(Error::Timer)
+            .and_then(move |_| {
+                web.eth().block_with_txs(BlockId::Number(BlockNumber::Latest)).map_err(Error::Source)
+            })
+            .filter_map(|maybe_block| maybe_block)
```

Comment:
:upside_down_face: Super nitpick but just to make sure that I understand things properly:

```suggestion
            .filter_map(|maybe_block| block)
```

Would that be more correct?

Reply:
Well, this would not compile :P 
But yes, `filter_map` filters out the `Option::None` elements in this case.

Reply:
Oh lol. Fixed my original suggestion :stuck_out_tongue:

Reply:
Well, the datatype is a `Option<Block>` and we `filter_map` so we make a `Block` out of an `Option<Block`, hence the name `maybe_block`. Makes sense? :)

Reply:
It makes sense, I had it wrong.